default.nix

```diff
@@ -0,0 +1,222 @@
+      content = builtins.readFile "${src}/include/page7.ps.inc";
+      dependencies = [ ];
+    };
+
+    checksumTable1 = {
+      content = builtins.readFile "${src}/include/checksum-table-1.ps.inc";
+      isLandscape = true;
+      dependencies = [ ];
+    };
+    checksumTable2 = {
+      content = builtins.readFile "${src}/include/checksum-table-2.ps.inc";
+      isLandscape = true;
+      dependencies = [ ];
+    };
+    checksumWorksheet = {
+      content = builtins.readFile "${src}/include/checksum-worksheet.ps.inc";
+      isLandscape = true;
+      dependencies = [ ];
+    };
+
+    shareTable = a: b: c: d: {
+      content = "${toString a} ${toString b} ${toString c} ${toString d} showShareTablePage\n";
+      dependencies = [ ];
+    };
+  };
+
+  fullBooklet = {
+    name = "SSS32.ps";
+    pages = with allPages; [
+      title
+      license
+      reference
+      principalTables
+      additionBottom
+      additionTop
+      generationInstructions
+      (shareTable 29 24 13 25)
+      (shareTable 9 8 23 18)
+      (shareTable 22 31 27 19)
+      (shareTable 1 0 3 16)
+      (shareTable 11 28 12 14)
+      (shareTable 6 4 2 15)
+      (shareTable 10 17 21 20)
+      (shareTable 26 30 7 5)
+      recovery
+      fusionInner
+      fusionOuter
+      checksumTable1
+      checksumTable2
+      checksumWorksheet
+    ];
+  };
+
+  dependencyContentRecur = content: builtins.concatMap
+    (item: (dependencyContentRecur item.dependencies) ++ [ item.content ])
+    content;
+  dependencyContent = pages: lib.lists.unique (
+    (map (dep: dep.content) standardDependencies) ++
+    (builtins.concatMap (page: dependencyContentRecur page.dependencies) pages)
+  );
+
+  renderBooklet = booklet:
+    let
+      addPage = content: pageData: {
+        content = content.content + lib.optionalString (pageData ? sourceHeader) ''
+          %****************************************************************
+          %*
+          %* ${pageData.sourceHeader}
+          %*
+          %****************************************************************
+        '' + ''
+          %%Page: ${toString content.nextPgIdx} ${toString content.nextPgIdx}
+          ${lib.optionalString (pageData ? isLandscape) "%%PageOrientation: Landscape\n"}%%BeginPageSetup
+          /pgsave save def
+          %%EndPageSetup
+        '' + (
+          if pageData ? drawPageContent
+          then
+            if pageData ? isLandscape
+            then "landscapePage begin ${toString content.nextCtIdx} drawPageContent\n"
+            else "portraitPage begin ${toString content.nextCtIdx} drawPageContent\n"
+          else
+            if pageData ? isLandscape
+            then "90 rotate\n"
+            else ""
+        ) + ''
+          ${pageData.content}
+          ${lib.optionalString (pageData ? drawPageContent) "end\n"}pgsave restore
+          showpage
+        '';
+        nextCtIdx = content.nextCtIdx + (if pageData ? drawPageContent then 1 else 0);
```

Comment:
Should `CtIdx` just be `PgIdx` - 1, or is that just a coincidence?

Reply:
It's just a coincidence, but let me check my current development branch to see if we want this calculation at all. It originates with the PostScript originally having a giant array of page content in the setup section, but now that we're assembling pages in Nix, I'm going to move the content into their respective pages rather than indexing into an array.

Regarding the page numbering logic, it's going to wind up being a little ugly, since we need to support

* Pages that get numbers
* Pages that are numbered, but the number isn't displayed because it'd interfere with the content (at least one volvelle page are like this)
* Pages that aren't numbered at all

I'd also like to support some preamble pages having numbers like `i`, `ii`, `iii`, etc.

Anyway for now maybe I just want to rip out the page numbering logic entirely since we need to iterate on it.